### ISSUE TITLE
Constant pointers to DataVariables recovered in LLIL don't properly update analysis when modified

### ISSUE THREAD

**Issue by joelreymont** [Vector35/binaryninja-api#4336]
**Version and Platform (required):**
- Binary Ninja Version: 3.5.4286-dev, 58311b01
- OS: macos
- OS Version: 13.3
- CPU Architecture: arm64

Consider the gigantic function at `0x6f20`. I don't know what is the proper way to call this data section below the function 

```
000073e4  void* data_73e4 = data_e2f98
000073e8  int32_t* data_73e8 = data_e2f84
000073ec  void* data_73ec = data_10c179
000073f0  void* data_73f0 = data_10c174
000073f4  void* data_73f4 = data_10c17e
000073f8  char const (* data_73f8)[0x6] = data_10c16f
```

but changing these to pointers does nothing to the function code above. 

```
000073ec  void* data_73ec = data_10c179
000073f0  void* data_73f0 = data_10c174
000073f4  void* data_73f4 = data_10c17e
```

I'm forced to eyeball the code and manually 'O' each use to convert them to pointers. 

This is extremely important because BN won't track the references to these otherwise! See #4317 for example.

The reason I bring that function in particular is because it's huge and the references are easy to miss. IDA does not have this issue and changing the data below the function is immediately reflected in the code above.

**Comment by joelreymont** [comment]
Anyone from V35 should search for "Encourage Salesman Prompt Delay" to find the database.

**Comment by xusheng6** [comment]
I am not getting this -- what are you expecting to happen? Changing the variable to a pointer does cause the function to be re-analyzed, though the final result is the same.

**Comment by joelreymont** [comment]
I expect BN to understand that these are pointers, track references and display the decompilation accordingly. 

Like IDA does.

**Comment by xusheng6** [comment]
> I expect BN to understand that these are pointers, track references and display the decompilation accordingly.
> 
> Like IDA does.

Unfortunately I still do not get it. Apart from that we initially failed to mark them as pointers, what are we missing after you set it to pointers?

**Comment by joelreymont** [comment]
May be fixed already, per #4317.
